web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,161 @@
+
+```python
+class BaseType:
+
+    @property
+    def dtype_backend -> Literal["pandas", "numpy", "pyarrow"]:
+        """
+        Library is responsible for the array implementation
+        """
+        ...
+
+    @property
+    def physical_type:
+        """
+        How does the backend physically implement this logical type? i.e. our
+        logical type may be a "string" and we are using pyarrow underneath -
+        is it a pa.string(), pa.large_string(), pa.string_view() or something else?
+        """
+        ...
+
+    @property
+    def missing_value_marker -> pd.NA|np.nan:
+        """
+        Sentinel used to denote missing values
+        """
+        ...
+```
+
+The theory behind this PDEP is that most users _should not_ care about the physical type that is being used. But if the abstraction our logical type provides is too much, a user could at least inspect and potentially configure which physical type to use.
+
+With regards to how we may expose such types to end users, there are two currently recognized proposals. The first would use factory functions to create the logical type, i.e. something like:
+
+```python
+pd.Series(["foo", "bar", "baz"], dtype=pd.string()  # assumed common case
+pd.Series(["foo", "bar", "baz"], dtype=pd.string(missing_value_marker=np.nan)
+pd.Series(["foo", "bar", "baz"], dtype=pd.string(physical_type=pa.string_view())
+```
+
+Another approach would be to classes:
+
+```python
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype()
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype(missing_value_marker=np.nan)
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype(physical_type=pa.string_view())
+```
+Note that the class-based approach would reuse existing classes like ``pd.StringDtype`` but change their purpose, whereas the factory function would more explicitly be a new approach. This is an area that requires more discussion amongst the team.
+
+## String Type Arguments
+
+This PDEP proposes that we maintain only a small subset of string arguments that can be used to construct logical types. Those string arguments are:
+
+  - intXX
+  - uintXX
+  - floatXX
+  - string
+  - datetime64[unit]
+  - datetime64[unit, tz]
+
+However, new code should be encouraged to use the logical constructors outlined previously. Particularly for aggregate types, trying to encode all of the information into a string can become unwieldy. Instead, keyword argument use should be encouraged:
+
+```python
+pd.Series(dtype=pd.list(value_type=pd.string()))
+```
+
+## Bridging Type Systems
+
+An interesting question arises when a user constructs two logical types with differing physical types. If one is backed by NumPy and the other is backed by pyarrow, what should happen?
+
+This PDEP proposes the following backends should be prioritized in the following order (1. is the highest priority):
+
+  1. Arrow
+  2. pandas
```

Comment:
maybe call this "masked"?  calling it "pandas" seems to preference it as being canonical

Reply:
Does `pandas extension` make more sense? Masked is maybe too specific, since this also refers to things like dictionary-encoded / categorical